trackstop.lua

```diff
@@ -0,0 +1,259 @@
+  local building = dfhack.gui.getSelectedBuilding()
+
+  building.friction = FRICTION_MAP[friction]
+end
+
+function TrackStopOverlay:getDumpDirection()
+  local building = dfhack.gui.getSelectedBuilding()
+  local use_dump = building.use_dump
+  local dump_x_shift = building.dump_x_shift
+  local dump_y_shift = building.dump_y_shift
+
+  if use_dump == 0 then
+    return NONE
+  else
+    if dump_x_shift == 0 and dump_y_shift == -1 then
+      return NORTH
+    elseif dump_x_shift == 1 and dump_y_shift == 0 then
+      return EAST
+    elseif dump_x_shift == 0 and dump_y_shift == 1 then
+      return SOUTH
+    elseif dump_x_shift == -1 and dump_y_shift == 0 then
+      return WEST
+    end
+  end
+end
+
+function TrackStopOverlay:setDumpDirection(direction)
+  local building = dfhack.gui.getSelectedBuilding()
+
+  if direction == NONE then
+    building.use_dump = 0
+    building.dump_x_shift = 0
+    building.dump_y_shift = 0
+  elseif direction == NORTH then
+    building.use_dump = 1
+    building.dump_x_shift = 0
+    building.dump_y_shift = -1
+  elseif direction == EAST then
+    building.use_dump = 1
+    building.dump_x_shift = 1
+    building.dump_y_shift = 0
+  elseif direction == SOUTH then
+    building.use_dump = 1
+    building.dump_x_shift = 0
+    building.dump_y_shift = 1
+  elseif direction == WEST then
+    building.use_dump = 1
+    building.dump_x_shift = -1
+    building.dump_y_shift = 0
+  end
+end
+
+function TrackStopOverlay:render(dc)
+  if not self:shouldRender() then
+    return
+  end
+
+  local building = dfhack.gui.getSelectedBuilding()
+  local friction = building.friction
+  local friction_cycle = self.subviews.friction
+
+  friction_cycle:setOption(FRICTION_MAP_REVERSE[friction])
+
+  self.subviews.dump_direction:setOption(self:getDumpDirection())
+
+  TrackStopOverlay.super.render(self, dc)
+end
+
+function TrackStopOverlay:shouldRender()
+  local building = dfhack.gui.getSelectedBuilding()
+  return building and building.trap_type == df.trap_type.TrackStop
+end
+
+function TrackStopOverlay:onInput(keys)
+  if not self:shouldRender() then
+    return
+  end
+  TrackStopOverlay.super.onInput(self, keys)
+end
+
+function TrackStopOverlay:init()
+  self:addviews{
+    widgets.Label{
+      frame={t=0, l=0},
+      text='Dump',
+    },
+    widgets.CycleHotkeyLabel{
+      frame={t=0, l=9},
+      key='CUSTOM_CTRL_X',
+      options={NONE, NORTH, EAST, SOUTH, WEST},
+      view_id='dump_direction',
+      on_change=function(val) self:setDumpDirection(val) end,
+    },
+    widgets.Label{
+      frame={t=1, l=0},
+      text='Friction',
+    },
+    widgets.CycleHotkeyLabel{
+      frame={t=1, l=9},
+      key='CUSTOM_CTRL_F',
+      options={NONE, LOW, MEDIUM, HIGH, MAX},
+      view_id='friction',
+      on_change=function(val) self:setFriction(val) end,
+    },
+  }
+end
+
+RollerOverlay = defclass(RollerOverlay, overlay.OverlayWidget)
+RollerOverlay.ATTRS{
+  default_pos={x=-71, y=29},
+  default_enabled=true,
```

Comment:
same here, since it's a gameplay mod, shouldn't be enabled by default and instead documented about how to turn on

Reply:
Removed the gameplay tag as discussed in another comment